toqito/states/mutually_unbiased_basis.py

```diff
@@ -0,0 +1,82 @@
+import numpy as np
+from sympy import primerange, isprime
+
+from toqito.matrices import gen_pauli
+
+
+def mutually_unbiased_basis(dim: int) -> list[np.ndarray]:
+    """
+    Function to generate list of MUBs for a given dimension.
+    
+    Note that this function only works if the dimension provided is prime or a power of a prime. Otherwise, we don't
+    know how to generate general MUBs.
+
+    Examples
+    ========
+    
+    For the case of dimension 2, the three mutually unbiased bases are provided by:
+
+    .. math::
+        M_0 = \left\{|0\rangle, |1\rangle \right\}, \\
+        M_1 = \left\{\frac{|0\rangle + |1\rangle}{\sqrt{2}}, \frac{|0\rangle - |1\rangle}{\sqrt{2}}\right\}
+        M_2 = \left\{\frac{|0\rangle + i|1\rangle}{\sqrt{2}}, \frac{|0\rangle - i|1\rangle}{\sqrt{2}}\right\}
+
+    References
+    ==========
+    .. [WikMUB] Wikipedia: Mutually unbiased bases
+        https://en.wikipedia.org/wiki/Mutually_unbiased_bases
+    
+    :param dim: The dimension of the mutually unbiased bases to produce.
+    :return: The set of mutually unbiased bases of dimension :code:`dim` (if known).
+    """
+    
+    # The first basis will always be the standard basis:
+    mats = [np.eye(dim)]
+
+    pauli_x = gen_pauli(1, 0, dim)
+    if isprime(dim):
+        pauli_z = gen_pauli(0, 1, dim)
+
+        for j in range(dim, 0, -1):
+            _, eigen_vec = np.linalg.eig(pauli_x @ pauli_z**(j))
+            mats.append(eigen_vec)
+    elif _is_prime_power(dim) and not isprime(dim):
+        raise ValueError(f"Dimension {dim} is a prime power but not prime (more complicated no support atm).")
+    else:
+        raise ValueError(f"No general construction of MUBs is known for dimension: {dim}.")
```

Comment:
```
        raise ValueError(f"No general construction of MUBs is known for dimension: {dim}.")
```
Might want to add a link to your repo in this function's docstring or in this `ValueError` message.  

https://github.com/vprusso/qubit-upb

Reply:
Ah, these are similarly named but different mathematical objects; this function is for mutually unbiased bases while the qubit UPB repo is unextendible product bases.